### PR TITLE
HW03

### PR DESCRIPTION
**Pull request by DanielHalachev**
Added source code and screenshots for HW03.

### CODE REVIEW

**Comment by DanielHalachev** [comment]
Note: I plan on grouping the source code into `src` and `include` directories, depending on what classes I make in the future. In any case, the source code will be organized in a better way.

File: SourceCode/Image.h [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#ifndef IMAGE_H
```
Comment (VladislavVulchevChaos) [inline SourceCode/Image.h:1]: You could use "#pragma once" instead

File: SourceCode/Ray.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#include "Ray.h"
+
+#include <cmath>
+
+Vector::Vector() : x(0), y(0), z(0){};
```
Comment (VladislavVulchevChaos) [inline SourceCode/Ray.cpp:5]: I prefer the {} constructor. Here you're using integer literals to initialize floats, the {} will warn you about that.

File: SourceCode/Ray.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  return {this->x - other.x, this->y - other.y, this->z - other.z};
+}
+
+Vector Vector::operator+(const Vector& other) const {
+  return {this->x + other.x, this->y + other.y, this->z + other.z};
+}
+
+void Vector::normalize() {
+  float length = this->length();
+  this->x = this->x / this->length();
```
Comment (VladislavVulchevChaos) [inline SourceCode/Ray.cpp:19]: You're caching the length, but do not use it afterwards, also you could go with only one division

File: SourceCode/Ray.h [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  unsigned short green;
+  unsigned short blue;
+  Color(unsigned short red, unsigned short green, unsigned short blue) : red(red), green(green), blue(blue){};
+  friend std::ostream &operator<<(std::ostream &os, const struct Color &color) {
+    os << color.red << " " << color.green << " " << color.blue;
+    return os;
+  }
+};
+
+struct Ray {
```
Comment (VladislavVulchevChaos) [inline SourceCode/Ray.h:27]: The Ray struct could be separated in its own file

File: SourceCode/Image.h [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+#define IMAGE_H
+struct Image {
+  unsigned int width;
+  unsigned int height;
```
Comment (VladislavVulchevChaos) [inline SourceCode/Image.h:5]: You could add the color buffer here also

**Pull request merged**